iwyu_path_util.cc

```diff
@@ -167,20 +178,33 @@ string MakeAbsolutePath(const string& path) {
 string MakeAbsolutePath(const string& base_path, const string& relative_path) {
   llvm::SmallString<128> absolute_path(base_path.c_str());
   llvm::sys::path::append(absolute_path, relative_path);
-
+  
   return absolute_path.str();
 }
 
 string GetParentPath(const string& path) {
   llvm::StringRef parent = llvm::sys::path::parent_path(path);
-  return parent.str();
+  string result = parent.str();
+  // Ensure trailing slash on non-empty paths
+  if(!result.empty() && result[result.size() - 1] != '/' && result[result.size() - 1] != '\\')
```

Comment:
I did not want to canonicalize the file path, as that would be beyond the scope of this function. Put in a backslash-path and get a slash-path although you only called `GetParent`?
And `parent.back()` is C++11